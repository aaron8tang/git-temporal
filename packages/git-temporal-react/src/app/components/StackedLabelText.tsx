import React from 'react';
import { style } from 'app/styles';

export interface StackedLabelTextProps {
  label: string;
  text: string;
  labelStyle?: object | string;
  textStyle?: object | string;
  onLabelClick?: (evt) => void;
}

const defaultContainerStyle = {
  flexGrow: 0,
  marginRight: 20,
  marginBottom: 10,
};

export const StackedLabelText = (props: StackedLabelTextProps): JSX.Element => {
  return (
    <div style={defaultContainerStyle}>
      <div style={style(props.labelStyle)} onClick={props.onLabelClick}>
        {props.label}
      </div>
      <div style={style(props.textStyle)}>{props.text}</div>
    </div>
  );
};
