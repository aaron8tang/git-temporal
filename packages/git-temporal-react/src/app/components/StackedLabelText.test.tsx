import React from 'react';
import { shallow } from 'enzyme';

import { StackedLabelText } from './StackedLabelText';

describe('components/StackedLabelText', () => {
  describe('when rendered with required props', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(
        <StackedLabelText label="This is the Label" text="this is the text" />
      );
    });

    test('it should match snapshot (it should have rendered a label and a text element)', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
