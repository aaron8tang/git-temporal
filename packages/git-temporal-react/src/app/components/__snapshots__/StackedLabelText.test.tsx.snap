// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/StackedLabelText when rendered with required props it should match snapshot (it should have rendered a label and a text element) 1`] = `
<div
  style={
    Object {
      "flexGrow": 0,
      "marginBottom": 10,
      "marginRight": 20,
    }
  }
>
  <div
    style={Object {}}
  >
    This is the Label
  </div>
  <div
    style={Object {}}
  >
    this is the text
  </div>
</div>
`;
